app/routes/user/profile_routes.py

```diff
@@ -84,6 +85,10 @@ def profile(username):
     # Set a flag indicating if this page view is for a user viewing another user's page
     viewing_self = user.username == current_user.username
 
+    # Check if user has set WCA ID to be public
+    show_wca_id_str = get_setting_for_user(user.id, SettingCode.SHOW_WCA_ID)
+    show_wca_id_bool = True if show_wca_id_str == 'true' else False
```

Comment:
You mentioned in a comment about turning this into a boolean, since the boolean setting returns a string 'true' or 'false'. All the settings are technically strings since I wanted to shoehorn them into the same table, and only have 1 column for setting value (not separate columns for boolean, string, etc).

For the sake of not using magic strings like this `'true'` here, I have a variable holding that in `settings_manager.py`: https://github.com/euphwes/cubers.io/blob/bc79f8bae6d888c24fd9755c17ce1dff33f3b893/app/persistence/settings_manager.py#L77

So for something like this, I'd prefer:

```python
from app.persistence.settings_manager import get_setting_for_user, SettingCode, TRUE_STR
...
show_wca_id_bool = show_wca_id_str == TRUE_STR
# or maybe even more concise...
show_wca_id_bool = get_setting_for_user(user.id, SettingCode.SHOW_WCA_ID) == TRUE_STR
```

I've also been chewing on tweaking the implementation of `get_setting_for_user` to check if the setting being returned is a boolean, and if so, just converting from string to boolean before returning. Should be an easy tweak but you'd just need to make sure that everywhere that retrieves boolean settings is aware that a bool is coming back, not a string. No biggie. If you wanted to merge this without making that change I'm ok with it, and I can just commit a little hotfix later.

Reply:
Thanks for the detailed comment on this PR 😃

What if we refactor ``get_setting_for_user`` like this?
```python
def get_setting_for_user(user_id, setting_code, as_python_bool=False):
    """ Retrieves a user's setting for a given setting code. """

    setting = DB.session.\
        query(UserSetting).\
        filter(UserSetting.user_id == user_id).\
        filter(UserSetting.setting_code == setting_code).\
        first()

    if as_python_bool:
        return True if setting.setting_value == TRUE_STR  else False
    else:
        return setting.setting_value if setting \
            else __create_unset_setting(user_id, setting_code).setting_value

```

So then we won't need to change other calls to this method, only the one I'm adding with this PR, which would become:
```python
show_wca_id_str = get_setting_for_user(user.id, SettingCode.SHOW_WCA_ID, as_python_bool=True)
```

Reply:
You bet!

I like the general idea, but then I think there's a risk of accidentally providing `as_python_bool=True` in a call for retrieving a non-boolean setting (like a string) and then it would always return `False`. 

I'd prefer to just check the setting itself to see if it's a boolean type, do the conversion, and then return the bool.

Maybe we compromise and do something in the middle, like this:

```python
# in settings_manager.py
def get_boolean_setting_for_user(user_id, setting_code):
    """ Retrieves a boolean setting for the user, and converts it to a bool type. """
    return get_setting_for_user(user_id, setting_code) == TRUE_STR

# in profile_routes
from app.persistence.settings_manager import get_boolean_setting_for_user, SettingCode
...
show_wca_id_bool = get_boolean_setting_for_user(user.id, SettingCode.SHOW_WCA_ID)
```

and then you don't need to change the method signature or update other calls right now. But the conversion to bool happens quietly in the settings manager and the profile route doesn't need to worry about it.

And then later, we can migrate the other boolean setting retrievals to use `get_boolean_setting_for_user` instead when we get around to it.

Reply:
That's a great solution. I'll implement this and push a new commit to this PR. Thanks for the elucidating ideas on the topic! As a beginner, this counts very much for me.

Reply:
You're welcome! I'm enjoying this from my side too, and it's nice to see other people's perspectives on the code.

Since this is just a personal project (though it's been running for a few years), it's a bit hacky and clunky in places, but it gets the job done. And stuff like is a great iterative way to improve the code base over time.